### ISSUE TITLE
Sorting on host search endpoints

### ISSUE THREAD

**Issue by ChrisSchinnerl** [SiaFoundation/renterd#1109]
-  `POST /autopilot/hosts`
    - ap score
-  `POST /autopilot/hosts` & `POST /bus/search/hosts`
    - rhp (whichever rhp version fields make sense)
      - storage remaining
      - storage price
      - download price
      - upload price
    - active contract
      - used storage
      - do we have used upload or download bandwidth?

Depends on https://github.com/orgs/SiaFoundation/projects/5/views/1?pane=issue&itemId=43563278

**Comment by alexfreska** [comment]
@ChrisSchinnerl will that dependent PR add all these sort options?

**Comment by ChrisSchinnerl** [comment]
No, it won't give you rhp2 or active contract info but more ways to filter on gouging and scoring info.

To get all of these will require a few more PRs on the backend which probably don't make sense to do before rhp4 since pricing is going to change again.
